bgpd/bgp_routemap.c

```diff
@@ -820,8 +821,11 @@ static route_map_result_t route_match_vni(void *rule,
 		if (path->extra == NULL)
 			return RMAP_NOMATCH;
 
-		if (vni == label2vni(&path->extra->label[0]))
-			return RMAP_MATCH;
+		for ( ; label_cnt < BGP_MAX_LABELS &&
```

Comment:
How are we constraining this to match only against VNIs (labels corresponding to a l2vpn/evpn route)? If this "match" were used in a route-map that is applied to labeled-unicast or ipxx-vpn, would the label there get matched?

Reply:
Thats a good question. For all 3 safi's MPLS_VPN, LABELED_UNICAST and EVPN, labels are stored in the same structure (bgp_path_info_extra). Hence, this match should work for these af's.

Reply:
You are right. Currently, this filter check based on vni can get executed for mpls and labeled unicast routes. I will make the change to limit this to vxlan tunnel type.